mlflow/projects/kubernetes.py

```diff
@@ -0,0 +1,110 @@
+import yaml
+import json
+import logging
+import docker
+import os
+import time
+import kubernetes
+from datetime import datetime
+
+_logger = logging.getLogger(__name__)
+
+
+def push_image_to_registry(image, registry, namespace, docker_auth_config):
+    docker_auth_config = json.loads(docker_auth_config)
+    repository = namespace + '/' + image
+    client = docker.from_env()
+
+    if registry:
+        repository = registry + '/' + repository
+        client.login(username=docker_auth_config['username'],
+                     password=docker_auth_config['password'],
+                     registry=registry)
+    else:
+        client.login(username=docker_auth_config['username'],
+                     password=docker_auth_config['password'])
+
+    client = docker.from_env()
+    image = client.images.get(name=image)
+    image.tag(repository)
+    client.images.push(repository=repository, auth_config=docker_auth_config)
+
+
+def _get_kubernetes_job_definition(image, image_namespace, job_namespace, command, env_vars):
+    timestamp = datetime.now().strftime('%Y-%m-%d-%H-%M-%S-%f')
+    enviroment_variables = ""
+    if os.environ.get('AZURE_STORAGE_ACCESS_KEY'):
+        env_vars['AZURE_STORAGE_ACCESS_KEY'] = os.environ['AZURE_STORAGE_ACCESS_KEY']
+    if os.environ.get('AZURE_STORAGE_CONNECTION_STRING'):
+        env_vars['AZURE_STORAGE_CONNECTION_STRING'] = os.environ['AZURE_STORAGE_CONNECTION_STRING']
+    for key in env_vars.keys():
+        enviroment_variables += "        - name: {name}\n".format(name=key)
+        enviroment_variables += "          value: \"{value}\"\n".format(value=env_vars[key])
+    job_template = (
+        "apiVersion: batch/v1\n"
+        "kind: Job\n"
+        "metadata:\n"
+        "  name: {job_name}-{timestamp}\n"
+        "  namespace: {job_namespace}\n"
+        "spec:\n"
+        "  template:\n"
+        "    spec:\n"
+        "      containers:\n"
+        "      - name: {container_name}\n"
+        "        image: {image_namespace}/{image_name}\n"
+        "        command: {command}\n"
+        "        env:\n"
+        "{enviroment_variables}"
```

Comment:
What about exposing the **resources.requests** and **resources.limits** as optional flags to the cli command? The user will expect to have control on this. 

Correct me if I'm wrong, the default values are based on the following order:
1. the namespace defaults - https://kubernetes.io/docs/tasks/administer-cluster/manage-resources/memory-default-namespace/
2. kubernetes defaults - https://cloud.google.com/blog/products/gcp/kubernetes-best-practices-resource-requests-and-limits

Reply:
hey @talgo10, that's a great idea. Actually I was working on a way to let users pass a path to a job template file. I've just pushed it, but I created the resources attribute in MLproject file too because I think this is an attribute that will be frequently used as you mentioned.